conn_js.go

```diff
@@ -0,0 +1,196 @@
+package websocket
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net"
+	"strings"
+	"sync"
+	"syscall/js"
+	"time"
+)
+
+const (
+	webSocketStateConnecting = 0
+	webSocketStateOpen       = 1
+	webSocketStateClosing    = 2
+	webSocketStateClosed     = 3
+)
+
+var errIsClosed = errors.New("connection is closed")
+
+// Conn implements net.Conn interface for WebSockets in js/wasm.
+type Conn struct {
+	js.Value
+	messageHandler *js.Func
+	closeHandler   *js.Func
+	mut            sync.Mutex
+	incomingData   chan []byte
+	currData       *bytes.Buffer
+	closeSignal    chan struct{}
+}
+
+func (c *Conn) Read(b []byte) (int, error) {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	if err := c.checkOpen(); err != nil {
+		return 0, io.EOF
```

Comment:
This should only return `io.EOF` if the connection was successfully closed. Is there any way to distinguish between some error condition and "closed"?

Reply:
`checkOpen` only returns an error if the connection is closed. I still need to finish handling errors in some cases (e.g. if the ["onerror" event](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/onerror) is triggered) but I will make sure we return any unexpected errors.